### ISSUE TITLE
User location missing from search result cards

### ISSUE THREAD

**Issue by jesseallhands** [Couchers-org/couchers#2121]
Multiple users have reported to the support team that they have received hosting requests from people who believe they are located in a different country than they are located in. This is because the current search results do not list the actual location of the  person (eg. city, country) and the search is based upon keyword (searching for "Rome" will show any user with "Rome" in their profile, eg. a user named "Romeo" or a person located in "Rome Corners, Wisconsin".)

To improve the user experience and avoid confusion we should include the user's location. This would save screen real-estate and would eliminate the need for a "show on map" button because when the user clicks on the location link, it would take them to the pin of the user on the map. Once the search functionality is improved to be based upon location, it will also help in situations where someone doesn't actually live in a city center but nearby (eg. someone searches "Chicago" and someone from "Naperville" shows up in the search results because they are nearby, but in reality Naperville is nearly an hour from downtown Chicago.)

**TLDR: Adding location to the search results cards would eliminate confusion which is currently happening, improve search UX and save screen real-estate.**

See mockup of what it would look like below:

![image](https://user-images.githubusercontent.com/79384268/138201865-859e3ae3-795b-4b32-a589-5d79454541aa.png)
